pkg/runtime/wfengine/wfengine_test.go

```diff
@@ -564,6 +565,68 @@ func TestRaiseEvent(t *testing.T) {
 	}
 }
 
+// TestPurge verifies that a workflow can have a series of activites created and then
+// verifies that all the metadata for those activities can be deleted from the statestore
+func TestPurge(t *testing.T) {
+	r := task.NewTaskRegistry()
+	r.AddOrchestratorN("ActivityChainToPurge", func(ctx *task.OrchestrationContext) (any, error) {
+		val := 0
+		for i := 0; i < 10; i++ {
+			if err := ctx.CallActivity("PlusOne", val).Await(&val); err != nil {
+				return nil, err
+			}
+		}
+		return val, nil
+	})
+	r.AddActivityN("PlusOne", func(ctx task.ActivityContext) (any, error) {
+		var input int
+		if err := ctx.GetInput(&input); err != nil {
+			return nil, err
+		}
+		return input + 1, nil
+	})
+
+	ctx := context.Background()
+	client, engine, stateStore := startEngineAndGetStore(ctx, r)
+	for _, opt := range GetTestOptions() {
+		t.Run(opt(engine), func(t *testing.T) {
+			id, err := client.ScheduleNewOrchestration(ctx, "ActivityChainToPurge")
+			if assert.NoError(t, err) {
+				metadata, err := client.WaitForOrchestrationCompletion(ctx, id)
+				if assert.NoError(t, err) {
+					assert.Equal(t, id, metadata.InstanceID)
+
+					// Get all the keys that were stored from the activity and ensure that they were stored
+					keysPrePurge := []string{}
+					keyCounter := 0
+					for key := range stateStore.(*daprt.FakeStateStore).Items {
+						keysPrePurge = append(keysPrePurge, key)
+						if strings.Contains(key, string(id)) {
+							keyCounter += 1
+						}
+					}
+					assert.Greater(t, keyCounter, 10)
+
+					err := client.PurgeOrchestrationState(ctx, id)
+					assert.NoError(t, err)
+
+					// Check that no key from the statestore containing the actor id is still present in the statestore
+					keysPostPurge := []string{}
+					for key := range stateStore.(*daprt.FakeStateStore).Items {
+						keysPostPurge = append(keysPostPurge, key)
+					}
+
+					for _, item := range keysPostPurge {
+						if strings.Contains(item, string(id)) {
+							assert.True(t, false)
+						}
+					}
+				}
+			}
+		})
+	}
+}
```

Comment:
Let's add one more test in order to ensure we're covering the multi-generation case. Basically, copy the `TestContinueAsNewWorkflow` logic and do a purge against that workflow state, validating that all state store content was successfully removed.